test/WebJobs.Script.Tests/ScriptHostTests.cs

```diff
@@ -1402,6 +1402,56 @@ public void ValidateFunction_ThrowsOnDuplicateName()
             Assert.Equal(string.Format($"The function or proxy name '{name}' must be unique within the function app.", name), ex.Message);
         }
 
+        [Fact]
+        public void ValidateFunction_ThrowsLegacyBlobTriggerOnFlexConsumption()
+        {
+            var httpFunctions = new Dictionary<string, HttpTriggerAttribute>();
+            var name = "test";
+
+            // first add an http function
+            var metadata = new FunctionMetadata();
+            metadata.Bindings.Add(new BindingMetadata()
+            {
+                Direction = BindingDirection.In,
+                Type = "blobTrigger",
+                Raw = JObject.Parse("{  \"type\": \"blobTrigger\",  \"connection\": \"\",  \"path\": \"sample1/{name}\",  \"name\": \"myBlob\"}")
+            });
+            var function = new Mock<FunctionDescriptor>(MockBehavior.Strict, name, null, metadata, null, null, null, null);
+            function.SetupGet(p => p.HttpTriggerAttribute).Returns(() => null);
```

Comment:
Seems this HttpTriggerAttribute setup shouldn't be needed?

Reply:
It's needed as we access it in `ValidateFunction`:
https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Host/ScriptHost.cs#L787